dvc/api.py

```diff
@@ -92,6 +94,175 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
         return fd.read()
 
 
+def get_params(
+    *targets: str,
+    repo: Optional[str] = None,
+    revs: Optional[Union[str, Iterable[str]]] = None,
+    deps: bool = False,
```

Comment:
This kwarg was not working as expected (https://dvc.org/doc/command-reference/params/diff#--deps) when multiple stages share the same `params` file since before the P.R. (haven't bisect yet to before the P.R.).

It's suppose to be useful to prevent loading `parameters` that are present in your params file but are not connected to your pipeline (not indicated as dependencies of any stage)

Reply:
Fixed the behavior in separate commit https://github.com/iterative/dvc/pull/7613/commits/fbe95c26ba839d15872355da6196f9d720ab9e8f as it was not really part of the new API added

Reply:
> I would expect params from all stages to be included.

I am taking a look.

> So it has no meaning when stages are provided, right?

Correct, `stages` implies `deps` but at a more granular level. I kept `deps` because it was already present but tbh not sure how valuable it is after supporting `stages`

Reply:
> I would expect params from all stages to be included.

It should work as expected now